Proposed issue template forms for bugs and feature requests

riverma opened the pull request:
- New GitHub template forms for bug reports and new feature requests
- Resolves: https://github.com/nasa/opera-sds-pge/issues/45 
- For live example, check out: https://github.com/riverma/github_test_area/issues/new/choose
- Both templates automatically tag issue with a new label "needs triage". _This label needs to be manually created once for the repository for this automatic tagging to work_

riverma commented:
@hhlee445 please review this as well - I'll propose the same patch to your PCM repos.

hhlee445 commented:
How can we assign an issue to the developer and set releases, and other information? Can it be done after the issue is created?

riverma commented:
> How can we assign an issue to the developer and set releases, and other information? Can it be done after the issue is created?

@hhlee445 - we can do that automatically in the issue creation or manually after. My recommendation is to do that manually after just like you'd do with any ticket someone else made. You'll notice that I've proposed the issue ticket templates be tagged automatically with a label "needs triage". I think the workflow after someone (or yourself) files a ticket is for the repo lead(s) would be:
1. (If filed by non leads) Find or be informed for tickets that "need triage" by filtering for issues in your repo
2. (If filed by non leads) Decide if ticket is worth doing
3. Assign to a developer and a particular release manually

hfattahi commented:
> 
Sounds great @riverma. @rtburns-jpl any opinion on these templates?

collinss-jpl reviewed (approved):
@riverma Looks great, thanks for jumping on this. FYI, I have added a "needs triage" label for this repo as you instructed.

rtburns-jpl commented:
I'd like to encourage plain text snippets over screenshots in the bug report template, as plain text is easier to read/search/interact with. Otherwise looks good to me 👍

riverma commented:
@collinss-jpl - thanks for adding the label. I've updated the PR with a commit to reflect @rtburns-jpl comments. 

If all looks good to you, feel free to merge for `opera-sds-pge` and I'll replicate the PR for other repos 👍 .

riverma commented:
Hey @collinss-jpl - I had to make one additional tweak to this PR, namely remove the repo-specific [line](https://github.com/nasa/opera-sds-pge/blame/main/.github/ISSUE_TEMPLATE/bug_report.yml#L14) pointing to my forked repo. Also I removed all checkboxes and converted them to dropdown so that GitHub doesn't create a task-list icon summary for issues when they're first made (the checkboxes weren't tasks, but confirmations).

I'll open another PR for these adjustments.

collinss-jpl commented:
> Hey @collinss-jpl - I had to make one additional tweak to this PR, namely remove the repo-specific [line](https://github.com/nasa/opera-sds-pge/blame/main/.github/ISSUE_TEMPLATE/bug_report.yml#L14) pointing to my forked repo. Also I removed all checkboxes and converted them to dropdown so that GitHub doesn't create a task-list icon summary for issues when they're first made (the checkboxes weren't tasks, but confirmations).
> 
> I'll open another PR for these adjustments.

@riverma Thanks for the heads up, it sounds like a pretty trivial change though, so in this instance I'm fine with you pushing the updates directly to `main` if you wish.
